sapi_demo/src/EventSubscriber/EntityViewEventSubscriber.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Drupal\sapi_demo\EventSubscriber;
+
+use Drupal\Component\Plugin\PluginManagerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\EventDispatcher\Event;
+use Drupal\Core\Session\AccountProxy;
+use Drupal\sapi\Dispatcher;
+use Drupal\Core\Routing\CurrentRouteMatch;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Drupal\sapi\ActionTypeInterface;
+use Drupal\Core\Entity;
+
+/**
+ * Class EntityViewEventSubscriber.
+ *
+ * @package Drupal\sapi_demo
+ */
+class EntityViewEventSubscriber implements EventSubscriberInterface {
+
+  /**
+   * Drupal\Core\Session\AccountProxy definition.
+   *
+   * @var \Drupal\Core\Session\AccountProxy
+   */
+  protected $currentUser;
+
+  /**
+   * Drupal\sapi\Dispatcher definition.
+   *
+   * @var \Drupal\sapi\Dispatcher
+   */
+  protected $sapiDispatcher;
+
+  /**
+   * The statistics action type plugin manager which will be used to create sapi
+   * items to be passed to the dispatcher
+   *
+   * @var \Drupal\Component\Plugin\PluginManagerInterface $sapiActionTypeManager
+   */
+  protected $sapiActionTypeManager;
+
+  /**
+   * Drupal\Core\Routing\CurrentRouteMatch definition.
+   *
+   * @var \Drupal\Core\Routing\CurrentRouteMatch
+   */
+  protected $currentRouteMatch;
+
+  /**
+   * Constructor.
+   */
+  public function __construct(AccountProxy $currentUser, Dispatcher $sapiDispatcher, PluginManagerInterface $sapiActionTypeManager, CurrentRouteMatch $currentRouteMatch) {
+    $this->currentUser = $currentUser;
+    $this->sapiDispatcher = $sapiDispatcher;
+    $this->sapiActionTypeManager = $sapiActionTypeManager;
+    $this->currentRouteMatch = $currentRouteMatch;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  static function getSubscribedEvents() {
+    // Priority is set to 1 to avoid this listener from being stopped by other
+    // listeners.
+    // @see ContainerAwareEventDispatcher::dispatch()
+    $events[KernelEvents::VIEW][] = ['onEventView', 1];
+    return $events;
+  }
+
+  /**
+   * Informs Statistics API dispatcher when controller outputs a value which is
+   * not a Response instance.
+   *
+   * @param \Symfony\Component\HttpKernel\Event\GetResponseForControllerResultEvent $event
+   */
+  public function onEventView(Event $event) {
+    try {
+      if (preg_match('/entity\.(?:[A-Za-z\-]+)\.canonical/',$this->currentRouteMatch->getRouteName())) {
+        /** @var $routeData[] An array of strings containing consecutive parts of route name. */
+        $routeData = explode('.',$this->currentRouteMatch->getRouteName());
```

Comment:
https://secure.php.net/manual/en/function.preg-match.php

>  matches
>    If matches is provided, then it is filled with the results of search. $matches[0] will contain the text that matched the full pattern, $matches[1] will have the text that matched the first captured parenthesized subpattern, and so on.

No need to do the explode if you are doing the pre_match.  Just create an array $matches, and pass it in as an argument in the preg_match, then check it's $matches[1]